pkg/daemon/rpm-ostree.go

```diff
@@ -196,6 +196,48 @@ func podmanInspect(imgURL string) (imgdata *imageInspection, err error) {
 
 }
 
+// skopeoInspect is a wrapper around the "skopeo inspect" command. Skopeo gets the proxy
+// from our environment, reads /etc/containers/registries.* so it should be mirror-aware
+// and it handles exponential backoff/retries that take the type of error into account. It's preferred
+// over podmanInspect because it doesn't have to pull the image to inspect it.
+func skopeoInspect(imgURL string) (imgdata *imageInspection, err error) {
+	// TODO(jkyros): This does not handle manifestlists. It looks like if you
+	// run with --raw, you can get the manifests and inspect them separately
+	// but we don't need that yet.
+
+	// Pull the container image if not already available
+	var authArgs []string
+	if _, err := os.Stat(kubeletAuthFile); err == nil {
+		authArgs = append(authArgs, "--authfile", kubeletAuthFile)
+	}
+
+	if !strings.HasPrefix(imgURL, "docker://") {
+		imgURL = "docker://" + imgURL
+	}
+
+	args := []string{"inspect", "--no-tags", "--retry-times", fmt.Sprintf("%d", numRetriesNetCommands)}
+	args = append(args, authArgs...)
+	args = append(args, imgURL)
+
+	var output []byte
+	// We let skopeo handle the retries, since it's smart enough to know what errors can be retried,
+	// If *we* do the retries, we're just blindly retrying every failure.
+	output, err = runGetOut("skopeo", args...)
```

Comment:
Do you remember what pulls in skopeo specifically into the MCD?

Reply:
it's part of RHCOS, we're freeloading off the host (I'm pretty sure just like podman) . We chroot way early in start.go so we should always have it.  https://github.com/openshift/machine-config-operator/blob/69933a3dccd6f78a5b9fe09d5b3830c9c8e39a03/cmd/machine-config-daemon/start.go#L125

That's maybe one reason why I might have favored the "use the code" approach vs "shell out" but I feel like we can always switch if it becomes a concern. There's plenty of precedent for switching ! :smile: